plugins/modules/openssl_pkcs12.py

```diff
@@ -360,10 +360,12 @@ def parse(self):
                 pkcs12_content = pkcs12_fh.read()
             p12 = crypto.load_pkcs12(pkcs12_content,
                                      self.passphrase)
-            pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM,
-                                          p12.get_privatekey())
-            crt = crypto.dump_certificate(crypto.FILETYPE_PEM,
-                                          p12.get_certificate())
+            pkey = p12.get_privatekey()
+            if pkey is not None:
+                pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM, pkey)
+            crt = p12.get_certificate()
+            if crt is not None:
+                crt = crypto.dump_certificate(crypto.FILETYPE_PEM, crt)
```

Comment:
Does this mean that the caller also has to check whether the pkey and crt returned from parse() is None?

It looks like Line 448-450 might need to check whether they are None before formatting their output:
   https://github.com/ansible-collections/community.crypto/pull/109/files#diff-f695e3d82cb39aaa7fe1f0598978dd25R448

Reply:
Yep, that one should better be adjusted as well, so the "parsed" file doesn't start with `NoneNone` :) Fixed that and added a test.